PhysicsTools/PatAlgos/plugins/PATTauProducer.cc

```diff
@@ -315,6 +315,11 @@ void PATTauProducer::produce(edm::Event & iEvent, const edm::EventSetup & iSetup
 	  edm::Handle<reco::PFTauDiscriminator> pfTauIdDiscr;
 	  iEvent.getByToken(pfTauIDTokens_[i], pfTauIdDiscr);
 
+	  if(!pfTauIdDiscr.isValid()){
```

Comment:
Thank you Michal.
Shouldn't the same protection be added also for CaloTau's below, at least for symmetry?

Reply:
OK, I will add this in a while even if I do not believe if anyone will use calo taus :)

Reply:
I think this is a really bad practice to "quietly" (nobody reads the warnings) skip on missing input.
This has to be handled at the configuration level.
Please add a configurable to possibly skip an unavailable input or better yet remove it explicitly from the relevant configuration.

Reply:
Dear Slava,
I think we are back to source of the discussion: if the product is not there an exception was produced by previous version. This avoided not-intended inconsistency between input sample and used configuration. In case the two were consistent, i.e. AOD produced with release new enough, the previous version worked. The problem was caused by an unexpected workflow setup used by unit tests. The issue is cured with current setup for a price of the "quiet" pass. Actually the "quiet" pass is probably what user could expect as in this way there is one-to-one correspondence between tau-Ids in AOD and MiniAOD.  
Regardless, I can add a configurable allowing/disallowing the "quiet" pass, but it will be set to True to pass the unit tests. This means that default setting will be also "quiet". Other solution, i.e. removal of the missing discriminator from configuration, will allow pass the tests, but will not allow to produce the discriminator for a fully consistent AOD->MiniAOD chain with one release which is not what is desired... 
The solution could be some customization, i.e. removal of the discriminant from the setup of "unrealistic" test - I think there is a way to have a customization for a particular PR, but I do not remember how to implement such a customization. 
Finally, I can also let the missing discriminant to be produced on the fly, but it will be different than other ones which are taken directly from AOD and not recomputed with the MVA new training. I can also produce all discriminants on the fly, but it will break AOD-miniAOD compatibility which is not what user can expect. 
Actually each solution has pros and cons and choice depends on what one defines as a required "default" behavior. 
It is already agreed that we will think on decoupling tau reco and Id for 10_0_X which will solve this kind of problems.

Reply:
> Regardless, I can add a configurable allowing/disallowing the "quiet" pass, but it will be set to True to pass the unit tests. This means that default setting will be also "quiet". 

please set default to not allow a quiet pass 
and apply a temporary customize to the unit tests that fail, with a clear comment there to clarify why it is there.
There should also be a github issue open together with this to revert this change.

There is no room to allow skipping of needed products in production just because some unit tests don't work and rely on just hopes that once we run in production this product will be there and useful for analysis.